docs/Developers/Integration/PTOracles.md

```diff
@@ -0,0 +1,106 @@
+
+## About Our Oracles
+
+Pendle's oracle implementation is inspired from the idea of UniswapV3 Oracle (see [here](https://docs.uniswap.org/concepts/protocol/oracle)) with a slight difference on how we define the cumulative rate. In short, our oracle stores the cumulative logarithm exchange rate between $PT$ and $Asset$ (please refer to the definition of assets [here](https://eips.ethereum.org/EIPS/eip-5115)).
+
+### EIP-5115 Asset
+
+Although the AMMs hold their positions and execute trades in $PT$ and $SY$, their underlying logic assumes all their $SY$ balances are converted to assets. As a result, the exchange rate stored in our oracles are between $PT$ and $Asset$, which should make pricing $PT$ more straightforward.
+
+Let's take `PT-stETH-25DEC2025` as an example. We have:
+1. $Asset$ in this case is `stETH`
+2. $oraclePtToAssetRate = 0.9$
+3. $stETHPrice = \$2000$ 
+
+Thus, the price of `PT-stETH-25DEC2025` should be $0.9 \times 2000 = \$1800$
+
+### Formulas
+
+Our oracle storage is in the following form:
+
+```sol
+struct Observation {
+    // the block timestamp of the observation
+    uint32 blockTimestamp;
+    // the tick logarithm accumulator, i.e., ln(exchangeRate) * time elapsed since the pool was first initialized
+    uint216 lnImpliedRateCumulative;
+    // whether or not the observation is initialized
+    bool initialized;
+}
+```
+
+The geometric mean price of $PT$ for the time interval of $[t_0, t_1]$ is:
+
+$$$
+lnPrice = \frac{lnImpliedRateCumulative_1 - lnImpliedRateCumulative_0}{t_1 - t_0}
+$$$
+$$$
+price = e^{lnPrice}
+$$$
+
+## Using PT as Collateral
+
+### Oracle Preparation
+
+Since our AMMs use TWAP oracles, firstly, we must define a $duration$ for the pricing interval. Same as UniswapV3 oracles, each $PT$ oracle has its own limitation on how far in the past the pricing data can be accessed. To check the validity of a duration for each a market, please query the following function from our `PendlePtOracle` contract:
+
+```sol
+function getOracleState(
+    address market,
+    uint32 duration
+)
+    external
+    view
+    returns (
+        bool increaseCardinalityRequired,
+        uint16 cardinalityRequired,
+        bool oldestObservationSatisfied
+    );
+```
+
+If `increaseCardinalityRequired` is returned `True`, it means the oracle's size of tracked observations must be expanded to `cardinalityRequired` before any price usage.
+
+If `oldestObservationSatisfied` is returned `False`, it means the oldest observations was recorded less than $duration$ seconds ago. In this case, the oracle needs more time to fill its observations to satisfy this specific $duration$.
+
+:::info
+
+To sum up, please make sure you have `increaseCardinalityRequired = False` and `oldestObservationSatisfied = True` before using our oracle on production.
+
+:::
+
+### Fetch price
```

Comment:
We should expand more on how people can get PT/USD price using PT/asset price, and elaborate more on the ChainlinkAsset oracle and GLP oracle. Maybe we can just add the example code snippet in here to explain how it works.